process/graduation_criteria.md

```diff
@@ -0,0 +1,115 @@
+- Remove obstacles to adoption and contribution by ensuring projects adhere to Microsoft Open Source Guidelines, code of
+  conduct and IP policy requirements
+  
+### Entry Requirements
+- Required backing of at least one member of the product management
+- Display of the AzCU "Sandbox" badge on website / readme
+- Draft project proposal for review and approval by TOC
+  - A great example is the [CNCF Project Proposal Process](https://github.com/cncf/toc/blob/fc161c13a86ba022277a906a10ff51a568406f7c/process/project_proposals.adoc)
+  - TODO: create our own project proposal process based on the CNCF project proposal process
+- Begin working toward achieving [Core Infrastructure Initiative Best Practices Badge](https://bestpractices.coreinfrastructure.org/)
+  - Must have 
+    - Gated PRs with unit tests
+    - MSFT code of conduct
+    - LCA bot enabled
+
+## Incubating Stage
+An Incubating project is one that has met all of the requirements of the Sandbox stage and is showing accelerating 
+adoption.
+
+### Entry Requirements
+- Minimum of 3 customers using the project in a production or pre-production environment
+- Minimum of 2 maintainers
+- Full due diligence of the TOC
+  - A great example is the [CNCF Due Diligence Guidelines](https://github.com/cncf/toc/blob/fc161c13a86ba022277a906a10ff51a568406f7c/process/due-diligence-guidelines.md)
+  - TODO: create our own DD guidelines based on the CNCF DD guidelines
+- Demonstrate a substantial ongoing flow of commits and merged contributions
+- Have a healthy number of issues and pull requests from external contributors
+- Set a clear, documented versioning scheme
+- Show significant progress toward achieving [Core Infrastructure Initiative Best Practices Badge](https://bestpractices.coreinfrastructure.org/)
+  - Must have 
+    - Unit tests and integration tests covering a majority of the codebase
+    - Linting
+    - Static analysis (where appropriate)
+    - An initial security review
+    - Contributing guide
+    - Issue and PR templates
+    - Public project board with prioritization
+    - Milestones for releases
+- Telemetry for errors and general usage
+  - If not possible, provide reasoning for why
+
+## Graduation Stage
+Graduation Stage project is one that has met all of the requirements of the Incubation stage and has reached a level
+of maturity and polish which will delight customers.
+
+### Entry Requirements
+- Minimum of 30 customers using the project in production
+- Minimum of 4 maintainers (at least one from outside AzCU)
+- Semester project road map published and updated
+- Full due diligence of the TOC with an eye to any miss alignment
+- Have achieved and maintained a [Core Infrastructure Initiative Best Practices Badge](https://bestpractices.coreinfrastructure.org/)
+  - Must have
+    - Load tests and performance tests
+  - Bonus for achieving Silver or higher
+- Full security review and threat analysis with the Azure Green / Red teams
```

Comment:
Yes. It is a requirement for MSFT QE process.

Reply:
Is the `full security review with the Azure Green / Red teams` going to give us the security audit mentioned in the CNCF graduation criteria doc? 

> Have completed an independent and third party security audit with results published of similar scope and quality as the following example (including critical vulnerabilities addressed): https://github.com/envoyproxy/envoy#security-audit and all critical vulnerabilities need to be addressed before graduation.

Reply:
You read my mind. That's precisely what I was hoping to achieve with a "third party" team. This is really the intent in the MSFT QE procedure as well. 

Basically, get some team with no stake in your project with a vast amount of security experience to perform a professional security review.

I'm not 100% sure that would cover the security audit requirement for CNCF, but I thought it would be pretty darn close.